apps/src/pythonlab/layout/ShareView.tsx

```diff
@@ -1,22 +1,102 @@
-import React from 'react';
+import Button from '@code-dot-org/component-library/button';
+import Toggle from '@code-dot-org/component-library/toggle';
+import React, {useEffect, useState} from 'react';
 
 import {hasPreview} from '@cdo/apps/codebridge';
+import Lab2Registry from '@cdo/apps/lab2/Lab2Registry';
+import pythonlabI18n from '@cdo/apps/pythonlab/locale';
 import {useAppSelector} from '@cdo/apps/util/reduxHooks';
+import commonI18n from '@cdo/locale';
 
 import ConsoleAndPreviewShare from './ConsoleAndPreviewShare';
 import ConsoleShare from './ConsoleShare';
 
 import moduleStyles from './share-view.module.scss';
 
+const SIDEBAR_WIDTH = 150;
+const MIN_PREVIEW_WIDTH = 200;
+
 const ShareView: React.FunctionComponent = () => {
   const miniApp = useAppSelector(
     state => state.lab2Project.projectSources?.labConfig?.miniApp?.name
   );
+  const projectManager = Lab2Registry.getInstance().getProjectManager();
+  const onViewCode = () => {
+    projectManager?.redirectToView();
+  };
+
+  const onRemix = () => {
+    projectManager?.redirectToRemix();
+  };
+  const [consoleVisible, setConsoleVisible] = useState(false);
+
+  const showPreview = hasPreview(miniApp);
+
+  const getAvailablePreviewWidth = () => {
+    return Math.max(window.innerWidth - SIDEBAR_WIDTH, MIN_PREVIEW_WIDTH);
+  };
+
+  const [previewWidth, setPreviewWidth] = useState(getAvailablePreviewWidth());
+  const [previewHeight, setPreviewHeight] = useState(window.innerHeight);
+
+  useEffect(() => {
+    window.addEventListener('resize', () => {
+      setPreviewWidth(getAvailablePreviewWidth());
+      setPreviewHeight(window.innerHeight);
+    });
+    return () =>
+      window.removeEventListener('resize', () => {
+        setPreviewWidth(getAvailablePreviewWidth());
+        setPreviewHeight(window.innerHeight);
+      });
+  }, []);
 
   return (
     <div className={moduleStyles.shareContainer}>
-      <div className={moduleStyles.sidebar}>Sidebar</div>
-      {hasPreview(miniApp) ? <ConsoleAndPreviewShare /> : <ConsoleShare />}
+      <div className={moduleStyles.sidebar} data-theme="Dark">
+        {showPreview && (
+          <div className={moduleStyles.consoleToggle}>
+            <Toggle
```

Comment:
If I click on toggle while neighborhood preview is running, I get an error. Not sure if we want to disable toggle until neighborhood preview run is completed...

https://github.com/user-attachments/assets/dcd6a451-819d-4694-a2f2-4921c9cb3f87

Reply:
Oh good call. I'll put that in the polish follow-up.